cloud/filestore/libs/vfs_fuse/fs_impl_data.cpp

```diff
@@ -491,21 +521,30 @@ void TFileSystem::FSyncDir(
     int datasync,
     fuse_file_info* fi)
 {
+    Y_ABORT_UNLESS(fi);
+
     STORAGE_DEBUG("FSyncDir #" << ino << " @" << fi->fh);
 
     if (!ValidateNodeId(*callContext, req, ino)) {
         return;
     }
 
+    if (!ValidateDirectoryHandle(*callContext, req, ino, fi->fh)) {
```

Comment:
уверен, что мы этой новой проверкой ничего не сломаем?

Reply:
this should be the same check that used in `readdir`  ... I can just ignore the fsyncdir and return in case of invalid directory file descriptor but it mean user app will get no error when using invalid fd which can lead to some bad behaviour

Reply:
ok, let's keep this validation